### PR TITLE
Add remote plugin GAV autocomplete, use model as fallback in projectbuildingerror

### PR DESCRIPTION
**Pull request by AObuchow**
add remote plugin GAV completion for #32

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

### CODE REVIEW

File: org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/MavenCompletionParticipant.java [hunk 1]
```diff
@@ -210,9 +210,9 @@ private void completeProperties(ICompletionRequest request, ICompletionResponse
 		allProps.put("project.groupId", project == null ? "unknown" : project.getGroupId());
 		allProps.put("project.artifactId", project == null ? "unknown" : project.getArtifactId());
 		allProps.put("project.name", project == null ? "unknown" : project.getName());
-		allProps.put("project.build.directory", project == null ? "unknown" : project.getBuild().getDirectory());
+		allProps.put("project.build.directory", project.getBuild() == null ? "unknown" : project.getBuild().getDirectory());
```
Comment (AObuchow) [inline org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/MavenCompletionParticipant.java]: There's already a null check that occurs for the project instance occuring beforehand, so we can assume project !=null

File: org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/MavenProjectCache.java [hunk 2]
```diff
@@ -121,6 +130,21 @@ private void parse(DOMDocument document) {
 				if (e.getCause() instanceof ModelBuildingException) {
 					ModelBuildingException modelBuildingException = (ModelBuildingException) e.getCause();
 					problems.addAll(modelBuildingException.getProblems());
+					try {
+						Model model = mavenReader.read(new FileReader(workingCopy));
+						MavenProject project = new MavenProject(model);
+						// TODO: use maven artifact repository which isin't deprecated
```
Comment (AObuchow) [inline org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/MavenProjectCache.java]: do this TODO

File: org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/MavenProjectCache.java [hunk 3]
```diff
@@ -121,6 +130,21 @@ private void parse(DOMDocument document) {
...
+						project.setRemoteArtifactRepositories(model.getRepositories().stream()
+								.map(repo -> new DefaultArtifactRepository(repo.getId(), repo.getUrl(), new DefaultRepositoryLayout() ))
+								.distinct().collect(Collectors.toList()));
+						project.setFile(workingCopy);
+						project.setBuild(new Build());
+						projectCache.put(uri, project);
+						projectParsedListeners.forEach(listener -> listener.accept(project));
+					} catch (IOException | XmlPullParserException e1) {
+						// TODO Auto-generated catch block
```
Comment (AObuchow) [inline org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/MavenProjectCache.java]: Remove

**Comment by mickaelistria** [comment]
I think it's OK to merge this as it and change it later if we find a better approach to handle incomplete models.

**Comment by AObuchow** [comment]
> 
> 
> I think it's OK to merge this as it and change it later if we find a better approach to handle incomplete models.

Okay sounds good

**Pull request merged**